draft-ietf-tls-esni.md

```diff
@@ -548,39 +562,61 @@ error code.
 
 ### HelloRetryRequest
 
-If the server sends a HelloRetryRequest in response to the ClientHello
-and the client can send a second updated ClientHello per the rules in
-{{RFC8446}}. At this point, the client does not know whether the
-server processed ClientHelloOuter or ClientHelloInner, and MUST
-regenerate both values to be acceptable. Note: if the inner and outer
-ClientHellos use different groups for their key shares or differ in
-some other way, then the HRR may actually be invalid for one or the
-other ClientHello. In that case, the Client MUST continue the
-handshake without changing the unaffected CH. Otherwise, the usual
-rules for HRR processing apply.
+If the server sends a HelloRetryRequest in response to the ClientHello,
+the client sends a second updated ClientHello per the rules in {{RFC8446}}.
+However, at this point, the client does not know whether the server processed
+ClientHelloOuter or ClientHelloInner, and MUST regenerate both values to
+be acceptable. Note: if the inner and outer ClientHellos use different groups
+for their key shares or differ in some other way, then the HRR may actually be
+invalid for one or the other ClientHello. In that case, the Client MUST continue
+the handshake without changing the unaffected CH. Otherwise, the usual rules for
```

Comment:
What does it mean to not change the outer ClientHello when the encrypted inner ClientHello is embedded inside it? What about binders?

Reply:
Good observation! I'll try to clarify. Basically, if the inner CH changes, any dependent state in the outer CH changes, too, including the "encrypted_client_hello" extension contents and any PSK binder.

Reply:
Okay, I attempted to clarify. Please have a look!